test/unit/stream_alert_cli/test_terraform_generate.py

```diff
@@ -0,0 +1,384 @@
+                    'logging_bucket': {
+                        'bucket': 'unit-testing.streamalert.s3-logging',
+                        'acl': 'log-delivery-write',
+                        'force_destroy': False,
+                        'versioning': {
+                            'enabled': True
+                        },
+                        'logging': {
+                            'target_bucket': 'unit-testing.streamalert.s3-logging',
+                            'target_prefix': 'unit-testing.streamalert.s3-logging/'
+                        },
+                        'lifecycle_rule': {
+                            'prefix': '/',
+                            'enabled': True,
+                            'transition': {
+                                'days': 30,
+                                'storage_class': 'GLACIER'
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        assert_equal(tf_main['provider'], tf_main_expected['provider'])
+        assert_equal(tf_main['terraform'], tf_main_expected['terraform'])
+        assert_equal(tf_main['resource'], tf_main_expected['resource'])
+
+    def test_generate_stream_alert(self):
+        """CLI - Terraform Generate stream_alert Module"""
+        pass
+
+    def test_generate_cloudwatch_monitoring(self):
+        """CLI - Terraform Generate cloudwatch_monitoring Module"""
+        cluster_name = 'test'
+        terraform_generate.generate_cloudwatch_monitoring(
+            cluster_name,
+            self.cluster_dict,
+            self.config
+        )
+
+        expected_cloudwatch_tf = {
+            'source': 'modules/tf_stream_alert_monitoring',
+            'sns_topic_arn': '${module.stream_alert_test.sns_topic_arn}',
+            'lambda_functions': [
+                'unit-testing_test_streamalert_rule_processor',
+                'unit-testing_test_streamalert_alert_processor'
+            ],
+            'kinesis_stream': 'unit-testing_test_stream_alert_kinesis'
+        }
+
+        assert_equal(
+            self.cluster_dict['module']['cloudwatch_monitoring_test'],
+            expected_cloudwatch_tf)
+
+    def test_generate_cluster_test(self):
+        """CLI - Terraform Generate 'Test' Cluster"""
+        config = self.config
+
+        tf_cluster = terraform_generate.generate_cluster(
+            config=config,
+            cluster_name='test'
+        )
+
+        cluster_keys = {
+            'module',
+            'output'
+        }
+
+        test_modules = {
+            'stream_alert_test',
+            'cloudwatch_monitoring_test',
+            'kinesis_test',
+            'kinesis_events_test'
+        }
+
+        assert_equal(set(tf_cluster['module'].keys()), test_modules)
+        assert_equal(set(tf_cluster.keys()), cluster_keys)
+
+    def test_generate_cluster_advanced(self):
+        """CLI - Terraform Generate 'Advanced' Cluster"""
+        config = self.config
```

Comment:
by reassigning to `config` here, you're essentially just creating a pointer to the `self.config` object anyway. omit this and just pass `config=self.config` in the `generate_cluster` call. the `self.config` is re-created in `setup` with each new method test anyway.